### PR TITLE
[PartEditor] Tidy registration

### PR DESCRIPTION
**Pull request by dayo09**
This commit tidies registration of PartEditorProvider.

ONE-vscode-DCO-1.0-Signed-off-by: dayo09 <dayoung.lee@samsung.com>

----

- For #1167

### CODE REVIEW

File: src/PartEditor/PartEditor.ts [hunk 1]
```diff
@@ -367,14 +367,18 @@ export class PartEditorProvider implements vscode.CustomTextEditorProvider, Part
 
   private _partEditors: PartEditor[] = [];
 
-  public static register(context: vscode.ExtensionContext): vscode.Disposable {
```
Comment (dayo09) [inline src/PartEditor/PartEditor.ts:370]: From [#1167](https://github.com/Samsung/ONE-vscode/issues/1167#issue-1329445337),

The function is tidy as is, but I am trying to standardize the registration function as below. 😄 

> Point 1. introduce `public static register(context: vscode.ExtensionContext) : void` function
> Point 2. Register each modules with the static function call
> ```
> OneTreeDataProvider.register(context);
> ```

File: src/extension.ts [hunk 2]
```diff
@@ -71,8 +71,8 @@ export function activate(context: vscode.ExtensionContext) {
 
   context.subscriptions.push(MondrianEditorProvider.register(context));
 
-  context.subscriptions.push(PartEditorProvider.register(context));
-  context.subscriptions.push(PartGraphSelPanel.register(context));
+  PartEditorProvider.register(context);
```
Comment (dayo09) [inline src/extension.ts:74]: From clean code's view, A little bit of improvement is that it does registration at one place. (not passing the provider disposable to the caller)

**Review by seanshpark — APPROVED** [review]
LGTM

**Review by jyoungyun — APPROVED** [review]
LGTM

**Pull request merged**